### PR TITLE
feat(Studio.Design): add presence

### PR DESCRIPTION
**Pull request by Milkeyyy**
## Description 
Add presence for the no-code website design tool, Studio.Design.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

studio design_preview

Studio Design_SC_1

Studio Design_SC_2

</details>

### CODE REVIEW

**Comment by DiscordAnaxes** [comment]
I'm not a presence reviewer on anything, but based on my experience, this presence would be treated as low-quality. This is due to the fact that it is entirely static as opposed to dynamic. Is there anything else in this website that could make it dynamic? Maybe like show what tools they are using, filename, etc. If you would like some suggestions, ask around in the PreMiD server I am sure someone can think of something creative.

**Review by theusaf — DISMISSED** [review]

File: websites/S/Studio.Design/presence.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		clientId: "1263373102746964009",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/zjBdwQ8.png",
+			startTimestamp: browsingTimestamp,
+		},
+		[privacy] = await Promise.all([presence.getSetting<boolean>("privacy")]),
```
Comment (darkvillager2) [inline websites/S/Studio.Design/presence.ts:11]: no?
```suggestion
		[privacy] = await (presence.getSetting<boolean>("privacy"))
```

File: websites/S/Studio.Design/presence.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		default:
+			// Project Dashboard
+			if (pathname.includes("/projects/") && pathname.includes("/dashboard/")) {
+				if (privacy) presenceData.details = "Project Dashboard";
+				else {
+					// Project Name
+					presenceData.details = `${
+						document.querySelector(
+							".LayoutPdDashboard > header > div._flex._gap-8._flex-initial._flex-row.o-layout-project-dashboard-header-left._mr-auto._h-full._py-6._pl-24 > div > button > div > p"
+						).textContent
```
Comment (darkvillager2) [inline websites/S/Studio.Design/presence.ts:34]:
```suggestion
					presenceData.details = privacy ? "Project dashboard" :  `${
						document?.title?.replace(" | Studio.Design", "")
```

File: websites/S/Studio.Design/presence.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		// Project Archive
+		case "/projects/archive": {
+			presenceData.details = "Browsing Project Archives...";
+			break;
+		}
+		default:
+			// Project Dashboard
+			if (pathname.includes("/projects/") && pathname.includes("/dashboard/")) {
+				if (privacy) presenceData.details = "Project Dashboard";
+				else {
```
Comment (darkvillager2) [inline websites/S/Studio.Design/presence.ts:29]:
```suggestion
```

File: websites/S/Studio.Design/presence.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+					// Project Name
+					presenceData.details = `${
+						document.querySelector(
+							".LayoutPdDashboard > header > div._flex._gap-8._flex-initial._flex-row.o-layout-project-dashboard-header-left._mr-auto._h-full._py-6._pl-24 > div > button > div > p"
+						).textContent
+					} | Dashboard`;
+				}
```
Comment (darkvillager2) [inline websites/S/Studio.Design/presence.ts:36]:
```suggestion
```

**Review by darkvillager2 — CHANGES_REQUESTED** [review]
Hello,

Thanks for opening this Pull Request.

I have a question about the code and i have requested changes to the code.

If you have any questions feel free to ask them here or on discord (@Dark_Ville)

**Comment by darkvillager2** [comment]
> I'm not a presence reviewer on anything, but based on my experience, this presence would be treated as low-quality. This is due to the fact that it is entirely static as opposed to dynamic. Is there anything else in this website that could make it dynamic? Maybe like show what tools they are using, filename, etc. If you would like some suggestions, ask around in the PreMiD server I am sure someone can think of something creative.

I agree that the quality isn't good, but its fixable.

File: websites/S/Studio.Design/presence.ts (hunk 2 above)
Comment (Milkeyyy) [reply websites/S/Studio.Design/presence.ts]: If the language setting is Japanese, the title is displayed as "Studio", so I fixed it using a regular expression.

**Review by darkvillager2 — APPROVED** [review]
LGTM

**Review by Bas950 — APPROVED** [review]

**Review by theusaf — APPROVED** [review]

**Pull request merged**